### PR TITLE
Add security context for fluenbit container

### LINKED ISSUE #793: help request: Need to pass the container level security context for FluentBit container
### Describe the issue

I need to pass  container level security context for FluentBit container via fluent Operator Helm chart. I am seeing in fluentbit CRD that there is no container level security context. Is there any way to pass the  security context for FluentBit container (container level) using Fluent Operator helm chart.

### How did you install fluent operator?

Helm Chart.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by karan56625**
### What this PR does / why we need it:

Adding container level security context for fluentbit container.

### Which issue(s) this PR fixes:

Fixes #793 

### Does this PR introduced a user-facing change?

```release-note

```

### Additional documentation, usage docs, etc.:

```docs

```

### CODE REVIEW

**Review by wenchajun — APPROVED** [review]

File: charts/fluent-operator/values.yaml [hunk 1]
```diff
@@ -91,6 +91,8 @@ fluentbit:
   secrets: []
   # Pod security context for Fluent Bit pods. Ref: https://kubernetes.io/docs/tasks/configure-pod-container/security-context/
   podSecurityContext: {}
+  # Security context for Fluent Bit container. Ref: https://kubernetes.io/docs/tasks/configure-pod-container/security-context/
+  securityContext: {}
```
[the remark this answers is not in the document]
Comment (karan56625) [reply charts/fluent-operator/values.yaml:95]: podSecurityContext -> for pod level security context
securityContext -> for container level security context.
just to make it consistent with what we have for fluent-operator pod.

`containerSecurityContext` is something I need to use internally, as containerContext was getting used for pod Security context internally.

**Review by adiforluls — APPROVED** [review]

File: charts/fluent-operator/values.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (adiforluls) [reply charts/fluent-operator/values.yaml:95]: Okay if this is what we have for fluent-operator container then LGTM

[the remark this answers is not in the document]
Comment (benjaminhuo) [reply charts/fluent-operator/values.yaml:95]: @adiforluls @karan56625 I think we're open to accepting any reasonable refactoring if you think it's necessary

[the remark this answers is not in the document]
Comment (karan56625) [reply charts/fluent-operator/values.yaml:95]: In that case, we should consider `securityContext` for container level security context and `podSecurityContext` for pod level security context  in Flunetbit CRD. (In helm chart, those fields seem fine, we can consider refactoring  Fluentbit CRD)

[the remark this answers is not in the document]
Comment (adiforluls) [reply charts/fluent-operator/values.yaml:95]: Ideally there shouldn't be a mismatch between CR fields and helm values from a UX perspective, but I think we'll need to rev the `apiVersion` of the CRD to maintain backwards compatibility, so to be pragmatic this is fine for now, when we do move to the next `apiVersion` we can refactor naming of certain fields (for example, `ClusterFluentBitConfig` is referenced as `fluentBitConfigName` in `FluentBit` CR).

**Pull request merged**